view/adminhtml/templates/element/navigation.phtml

```diff
@@ -0,0 +1,107 @@
+<?php
+/** @var \Magento\Framework\Escaper $escaper */
+/** @var \Hyva\Admin\Block\Grid\Navigation $block */
+$parent = $block->getParent();
+$grid = $parent->getGrid();
+$navigation = $grid->getNavigation();
```

Comment:
Please assign the navigation view model directly as a template variable from the view model, instead of fetching it from the grid via the parent block.
See `\Hyva\Admin\ViewModel\HyvaGrid\Cell::createTemplateBlock` for an example.

Reply:
I know what you mean, but when creating the templates I do not know its purpose.
I think it is in terms of extensibility better to assign the grid and each component that is defined in the di.xml then gets the grid provided. I think what I did here could be a first approach to split the grid.phtml and make it more extendable.
What do you think.